ethcore/src/client/client.rs

```diff
@@ -1692,6 +1692,9 @@ impl BlockChainClient for Client {
 		if let Some(after) = after {
 			if let Err(e) = iter.seek(after) {
 				trace!(target: "fatdb", "list_accounts: Couldn't seek the DB: {:?}", e);
+			} else {
+				// Position the iterator after the `after` element
+				iter.next();
```

Comment:
Could you explain how this works? Why calling `next` after `seek` returned an `Err` would position the iterator after the `after`?

Reply:
So the `else` branch is actually when there has been no errors

Reply:
Ah, right, thanks for clarifying that. So we assume that `after` is present in the db, otherwise this is not correct?

Reply:
So in this context, `last` is whether `None` or an address that is actually in DB.

Reply:
I think this could be simplified by writing:

```rust
if let Err(e) = iter.next(after) {
  trace!("...");
}
```

Cause `Err` is no used later on anyway